eoscostarica/eos-rate
# Add kubernetes deployment

### GH Issue

#  improvement: add kubernetes deployment and service files (#375)
    
    * Use NFS storage instead of a local directory for
    volumes.
    * Allocate external IPs for exposing services.
    * Use a docker registry so we don't have to build the containers
      manually on each worker node.
    
    Still TODO:
    * Use Openstack as external cloud provider (cinder and neutron).
    
    refs #224 and #https://github.com/edenia/openstack-docs/issues/18.
        depends on the frontend container created in #366.


### Steps to test

1. kubectl create -f kubernetes/
1. kubectl expose deployment frontend --name=frontend-port --port=80 --target-port=80 --type=ClusterIP --external-ip={external_ip}
1. visit {external_ip}
1. you should see the eos rate page

### Checklist
